src/cargo/util/toml/mod.rs

```diff
@@ -2151,6 +2225,41 @@ fn to_dependency_source_id<P: ResolveToPath + Clone>(
     }
 }
 
+pub(crate) fn lookup_path_base<'a>(
+    base: &PathBaseName,
+    gctx: &GlobalContext,
+    workspace_root: &dyn Fn() -> CargoResult<Option<&'a PathBuf>>,
+    features: &Features,
+) -> CargoResult<PathBuf> {
+    features.require(Feature::path_bases())?;
+
+    // Look up the relevant base in the Config and use that as the root.
+    // NOTE: The `base` string is user controlled, but building the path is safe from injection
+    // attacks since the `PathBaseName` type restricts the characters that can be used.
+    if let Some(path_bases) =
+        gctx.get::<Option<ConfigRelativePath>>(&format!("path-bases.{base}"))?
```

Comment:
imo the structure makes this harder to follow
```suggestion
    // HACK: The `base` string is user controlled, but building the path is safe from injection
    // attacks since the `PathBaseName` type restricts the characters that can be used to exclude `.`
    let base_key = format!("path-bases.{base}");
    
    // Look up the relevant base in the Config and use that as the root.
    if let Some(path_bases) =
        gctx.get::<Option<ConfigRelativePath>>(&base_key)?
```

Reply:
ok